nilearn/datasets/utils.py

```diff
@@ -774,6 +766,85 @@ def _fetch_files(data_dir, files, resume=True, mock=False, verbose=1):
     return files_
 
 
+def _message_or_mock_missing_dataset_file(file_, mock, abort, dl_file, target_file,
+                                          temp_target_file, url):
+    """ Warns & returns an abort message if the downloaded dataset does not have the intended file.
+    If mock==True (intended for testing), creates a mock file instead.
+    Returns
+    -------
+    abort: None or String
+    """
+    
+    if mock:
+        if not os.path.exists(os.path.dirname(temp_target_file)):
+            os.makedirs(os.path.dirname(temp_target_file))
+        open(temp_target_file, 'w').close()
+    else:
+        warnings.warn('An error occured while fetching %s' % file_)
+        abort = ("Dataset has been downloaded but requested file was "
+                 "not provided:\nURL: %s\n"
+                 "Target file: %s\nDownloaded: %s" %
+                 (url, target_file, dl_file))
+    return abort
+
+
+def _check_files_existence(file_1, file_2):
+    """ Returns True if both the provided files exist.
+    """
+    target_file_does_not_exist = not os.path.exists(file_1)
+    temp_target_file_does_not_exist = not os.path.exists(file_2)
+    file_does_not_exist = target_file_does_not_exist and temp_target_file_does_not_exist
+    return file_does_not_exist
```

Comment:
3-liner functions are in general bad practice. The reason being that when doing a careful read of the codebase, one needs to read them to know what they actually do (the name is not a contract, and bugs hide this).

I am not very enthousiastic about merging changes that turn 3 lines of code used twice into a function. It makes the code harder to read and maintain.